app/src/main/java/io/homeassistant/companion/android/settings/qs/views/ManageTilesView.kt

```diff
@@ -0,0 +1,144 @@
+                        viewModel.selectedIcon.value = viewModel.currentTile()!!.iconId
+                        val loader = IconPackLoader(context)
+                        iconPack = createMaterialDesignIconPack(loader)
+                        iconPack.loadDrawables(loader.drawableLoader)
+                        val iconDrawable = viewModel.selectedIcon.value?.let { iconPack.getIcon(it)?.drawable }
+                        if (iconDrawable != null) {
+                            val icon = DrawableCompat.wrap(iconDrawable)
+                            icon.setColorFilter(context.resources.getColor(io.homeassistant.companion.android.R.color.colorAccent), PorterDuff.Mode.SRC_IN)
+                            viewModel.icon.value = icon
+                        }
+                    }
+                }) {
+                    Text(tileName)
+                }
+            }
+        }
+        Text(
+            text = stringResource(id = R.string.tile_label),
+            modifier = Modifier.padding(start = 10.dp, bottom = 10.dp)
+        )
+        TextField(
+            value = viewModel.tileLabel.value.toString(),
+            onValueChange = { viewModel.tileLabel.value = it },
+            modifier = Modifier.padding(start = 10.dp, bottom = 10.dp)
+        )
+        Text(
+            text = stringResource(id = R.string.tile_subtitle),
+            modifier = Modifier.padding(start = 10.dp, bottom = 10.dp)
+        )
+        TextField(
+            value = viewModel.tileSubtitle.value.toString(),
+            onValueChange = { viewModel.tileSubtitle.value = it },
+            modifier = Modifier.padding(start = 10.dp, bottom = 10.dp)
+        )
+
+        TextButton(onClick = {
+            iconDialog.show(childFragment, viewModel.selectedTile.value)
+        }) {
+            Text(
+                text = stringResource(id = R.string.tile_icon),
+                fontSize = 15.sp,
+                modifier = Modifier.padding(end = 10.dp)
+            )
+            viewModel.icon.value?.toBitmap()?.asImageBitmap()
+                ?.let { Image(it, contentDescription = stringResource(id = R.string.tile_icon)) }
+        }
+        TextButton(onClick = { expandedEntity = true }) {
+            Text(
+                text = stringResource(id = R.string.tile_entity) + ": ${viewModel.selectedEntityId.value}",
+                fontSize = 15.sp
+            )
+        }
+        DropdownMenu(expanded = expandedEntity, onDismissRequest = { expandedEntity = false }) {
+            val sortedEntities = viewModel.entities.values.sortedBy { it.entityId }
+            for (item in sortedEntities) {
+                DropdownMenuItem(onClick = {
+                    viewModel.selectedEntityId.value = item.entityId
+                    expandedEntity = false
+                }) {
+                    Text(text = item.entityId, fontSize = 15.sp)
+                }
+            }
+        }
+        TextButton(onClick = {

```

Comment:
Is this button really necessary or could the app also update the database (and push it to the system) when any of the input fields change?
If it is necessary: this button should be emphasized to make sure the user actually taps on it and doesn't quit without saving. Suggestion: use a regular, full-width `Button`.

Reply:
So this button was designed to ensure the user is actually changing the tile they meant to change. If we make it autoupdate when any field is changed it has the potential of overwriting data a user did not expect. I will make this a button to make it more emphasized :)